modules/packages/LinearAlgebra.chpl

```diff
@@ -1857,16 +1857,17 @@ module Sparse {
 
   /* Transpose CSR domain */
   proc transpose(D: domain) where isCSDom(D) {
-    var indices: [1..0] 2*D.idxType;
+    use Lists;
+    var indices: list(2*D.idxType);
     for i in D.dim(1) {
       for j in D.dimIter(2, i) {
-        indices.push_back((j, i));
+        indices.append((j, i));
       }
     }
 
     const parentDT = transpose(D.parentDom);
     var Dom: sparse subdomain(parentDT) dmapped CS(sortedIndices=false);
-    Dom += indices;
+    Dom += indices.toArray();
```

Comment:
More as a note going forward, but this makes performance of this method even worse then it was before. I am hoping to make this implementation more elegant and perform better with this: https://github.com/chapel-lang/chapel/pull/13420

Reply:
Do you think there's anything that can be done to soften the performance hit, short of just using the deprecated features?

Reply:
We can just iterate over `indices` and add each to `Dom` individually, right?

In any case, see what the now-replaced += did, and do the same here.

Reply:
Right, that's also an option.

`[i in indices] Dom += i;` should work, but it is hard to say whether that'd be faster or slower than this approach.

Reply:
I would advocate for @dlongnecke-cray not spending any additional effort modifying this implementation and accepting a temporary performance hit, with the assumption that @e-kayrakli will get around to writing the improved version with buffered index addition within the release cycle.

Reply:
@ben-albrecht -- totally agree